### DISCUSSION TITLE
Installing MET on a new computer system

### DISCUSSION THREAD

**Discussion by jcorner1** [dtcenter/METplus#2010] [Installation] [answered]
Hello,

I'm back to installing met again on a new computer system in house at my university. Seem to have gotten a fair bit done already, but unsure how to proceed now as I'm having a few issues. I will attach my files below. Thanks for the help!

[configure.txt](https://github.com/dtcenter/METplus/files/10443202/configure.txt)
[config.txt](https://github.com/dtcenter/METplus/files/10443204/config.txt)
[compile_MET_all.sh.txt](https://github.com/dtcenter/METplus/files/10443208/compile_MET_all.sh.txt)
[install_met_env.triton.txt](https://github.com/dtcenter/METplus/files/10443209/install_met_env.triton.txt)

**Comment by jprestop** [comment, marked as answer]
Hi @jcorner1.  Thanks so much for including these files!  

I see the following in the config.log file:
```
/usr/bin/ld: cannot find -lhdf5_hl
/usr/bin/ld: cannot find -lhdf5
/usr/bin/ld: cannot find -lz
collect2: error: ld returned 1 exit status
```

So it looks like MET cannot find the library files for HDF5 and the Z library.  I don't see values for those set in your install_met_env.triton configuration file, so it looks like you intended for the script to build those for you.  Do you happen to have the screen output available from when you ran the script that you could pass along?  You could check to see if those libraries were successfully installed by looking at the contents of /home/jcorner1/Masters/external_libs/lib and see if those library files exist.  If not, something when wrong in the install process with those files and we'll need to take a look to see where the script failed and why they were not installed.

**Reply by jprestop** [reply]
Hello again @jcorner1.

On a separate note, looking ahead to check on some Python values, I see that you have the following set in your install_met_env.triton file:
```
#Python ldflags created using python3-config
export MET_PYTHON_LD="$(python3-config --ldflags)"

#Python cflags created using python3-config
export MET_PYTHON_CC="$(python3-config --cflags)"
```

Those result in the following values:
```
MET_PYTHON_CC='-I/anaconda3/include/python3.9 -I/anaconda3/include/python3.9  -Wno-unused-result -Wsign-compare -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O3 -ffunction-sections -pipe -isystem /anaconda3/include -fdebug-prefix-map=/opt/conda/conda-bld/python-split_1649141344976/work=/usr/local/src/conda/python-3.9.12 -fdebug-prefix-map=/anaconda3=/usr/local/src/conda-prefix -I/anaconda3/include -DNDEBUG  -O3 -Wall'
MET_PYTHON_LD='-L/anaconda3/lib/python3.9/config-3.9-x86_64-linux-gnu -L/anaconda3/lib  -lcrypt -lpthread -ldl  -lutil -lm -lm '
```
which unfortunately is not quite what we need.  _python3-config --cflags_ is returning too much information, and _python3-config --ldflags_ is not returning enough information.

Please reset the values in your install_met_env.triton file to be instead:
```
export MET_PYTHON_LD="-L/anaconda3/lib/python3.9/config-3.9-x86_64-linux-gnu -L/anaconda3/lib  -l python3.9 -lcrypt -lpthread -ldl  -lutil -lm"
export MET_PYTHON_CC=-I/anaconda3/include/python3.9 -I/anaconda3/include/python3.9
```

**Reply by jcorner1** [reply]
Hi @jprestop,

Thanks for your help once again! here is the screen output when I run the files. I also changed the values indicated, so I will send all my files as well.

```
[jcorner1@triton met]$ ./compile_MET_all.sh install_met_env.triton 
install_met_env.triton: line 43: export: `-I/anaconda3/include/python3.9': not a valid identifier

TEST_BASE = /home/jcorner1/Masters/met/MET-11.0.0
COMPILER  = gnu
MET_SUBDIR = /home/jcorner1/Masters/met/MET-11.0.0
MET_TARBALL = MET-11.0.0.tar.gz
USE_MODULES = FALSE
LD_LIBRARY_PATH = /home/jcorner1/Masters/met/MET-11.0.0/external_libs/lib:/anaconda3/lib:/home/jcorner1/Masters/met/MET-11.0.0/external_libs/lib/lib:/home/jcorner1/Masters/met/MET-11.0.0/external_libs/lib:/home/jcorner1/Masters/met/MET-11.0.0/external_libs/lib

Compiling libraries into: /home/jcorner1/Masters/met/MET-11.0.0/external_libs

USE_MODULES = FALSE

export  CC=/usr/bin/gcc
export CXX=/usr/bin/g++
export  FC=/usr/bin/gfortran
export F77=/usr/bin/gfortran
export F90=/usr/bin/gfortran

Using python version: 
Python 3.9.12

Compiling GSL_2.1 at Thu Jan 19 16:44:42 UTC 2023
cd /home/jcorner1/Masters/met/MET-11.0.0/external_libs/gsl/gsl-2.1
./configure --prefix=/home/jcorner1/Masters/met/MET-11.0.0/external_libs > configure.log 2>&1
make > make.log 2>&1
make install > make_install.log 2>&1

Compiling MET at Thu Jan 19 16:50:16 UTC 2023
./compile_MET_all.sh: line 799: cd: MET*: No such file or directory
MET Configuration settings...
export MET_FONT_DIR=/home/jcorner1/Masters/met/MET-11.0.0/fonts
export MET_GRIB2CLIB=/home/jcorner1/Masters/met/MET-11.0.0/external_libs/lib
export MET_GSL=/home/jcorner1/Masters/met/MET-11.0.0/external_libs
export MET_PYTHON_CC=-I/anaconda3/include/python3.9
export MET_TARBALL=/home/jcorner1/Masters/met/MET-11.0.0/tar_files/MET-11.0.0.tar.gz
export MET_GRIB2CINC=/home/jcorner1/Masters/met/MET-11.0.0/external_libs/include
export MET_NETCDF=/home/jcorner1/Masters/met/MET-11.0.0/external_libs/lib
export MET_PYTHON=/anaconda3
export MET_PYTHON_LD=-L/anaconda3/lib/python3.9/config-3.9-x86_64-linux-gnu -L/anaconda3/lib  -l python3.9 -lcrypt -lpthread -ldl  -lutil -lm
export MET_SUBDIR=/home/jcorner1/Masters/met/MET-11.0.0
export MET_BUFRLIB=/home/jcorner1/Masters/met/MET-11.0.0/external_libs/lib
LDFLAGS = -Wl,--disable-new-dtags -Wl,-rpath,/home/jcorner1/Masters/met/MET-11.0.0/external_libs/lib:/home/jcorner1/Masters/met/MET-11.0.0/external_libs/lib/lib:/home/jcorner1/Masters/met/MET-11.0.0/external_libs/lib:/home/jcorner1/Masters/met/MET-11.0.0/external_libs/lib:/anaconda3/lib:/home/jcorner1/Masters/met/MET-11.0.0/external_libs/lib -Wl,-rpath,   
cd /home/jcorner1/Masters/met/MET-11.0.0
./configure --prefix=/home/jcorner1/Masters/met/MET-11.0.0 --bindir=/home/jcorner1/Masters/met/MET-11.0.0/bin BUFRLIB_NAME=-lbufr GRIB2CLIB_NAME=-lgrib2c --enable-grib2 --enable-python  > configure.log 2>&1
configure returned with non-zero (77) status
[jcorner1@triton met]$ 
```


[install_met_env.triton (2).txt](https://github.com/dtcenter/METplus/files/10459169/install_met_env.triton.2.txt)
[configure (1).txt](https://github.com/dtcenter/METplus/files/10459170/configure.1.txt)
[config (1).txt](https://github.com/dtcenter/METplus/files/10459171/config.1.txt)
[compile_MET_all.sh (2).txt](https://github.com/dtcenter/METplus/files/10459172/compile_MET_all.sh.2.txt)

**Reply by jprestop** [reply]
Thank you for this helpful information.  First, my apologies, I see I made in error in asking you set MET_PYTHON_CC to "-I/anaconda3/include/python3.9 -I/anaconda3/include/python3.9".  I didn't initially notice there were two "-I" followed by the paths.  Could you please instead set:
```
export MET_PYTHON_CC=-I/anaconda3/include/python3.9
```

Could you please also grab our latest version of the compile_MET_all.sh script, which was just updated yesterday, by running the following?
```
wget https://raw.githubusercontent.com/dtcenter/MET/main_v11.0/internal/scripts/installation/compile_MET_all.sh
```
You may also need to change the permissions to make it executable (but I'm not 100% positive you will need to).  You can do that by running the following:
```
chmod 775 compile_MET_all.sh
```

Then, please remove this value from your configuration file:
```
export MET_NETCDF=/home/jcorner1/Masters/met/MET-11.0.0/external_libs/lib
```
That will allow HDF5 and NetCDF4 to be recompiled. If we get a successful compilation of HDF5 and NetCDF, but then something fails later, you'd be able to reset that value (along with setting a value for MET_HDF5, however, you'd want to set it to the value above, but remove the "/lib".  This is because the MET code will set MET_NETCDFLIB (/home/jcorner1/Masters/met/MET-11.0.0/external_libs/lib) and MET_NETCDFINC (/home/jcorner1/Masters/met/MET-11.0.0/external_libs/include) based on the base value for MET_NETCDF of /home/jcorner1/Masters/met/MET-11.0.0/external_libs.

And one final request, after completing the above, could you please send me the output of running the following?
```
ls /home/jcorner1/Masters/met/MET-11.0.0/external_libs/lib
```

Thanks!

**Reply by jcorner1** [reply]
hi @jprestop,

Here is the output from running the list command:

```
[jcorner1@triton met]$ ls /home/jcorner1/Masters/met/MET-11.0.0/external_libs/lib
libgsl.a  libgslcblas.a  libgslcblas.la  libgslcblas.so  libgslcblas.so.0  libgslcblas.so.0.0.0  libgsl.la  libgsl.so  libgsl.so.19  libgsl.so.19.0.0  pkgconfig
```
although after running everything with the new changes I got:

```
[jcorner1@triton met]$ ls /home/jcorner1/Masters/met/MET-11.0.0/external_libs/lib
libgsl.a       libgslcblas.la  libgslcblas.so.0      libgsl.la  libgsl.so.19      libhdf5.a     libhdf5_hl.la  libhdf5_hl.so.200      libhdf5.la        libhdf5.so      libhdf5.so.200.2.0
libgslcblas.a  libgslcblas.so  libgslcblas.so.0.0.0  libgsl.so  libgsl.so.19.0.0  libhdf5_hl.a  libhdf5_hl.so  libhdf5_hl.so.200.1.0  libhdf5.settings  libhdf5.so.200  pkgconfig
```
I was unsure of what ls command you wanted to send, so I sent both.

and here are the files I have been sending. I do believe the new compile file is in there.

[compile_MET_all.txt](https://github.com/dtcenter/METplus/files/10460452/compile_MET_all.txt)
[install_met_env.triton.txt](https://github.com/dtcenter/METplus/files/10460453/install_met_env.triton.txt)
[configure.txt](https://github.com/dtcenter/METplus/files/10460454/configure.txt)
[config.txt](https://github.com/dtcenter/METplus/files/10460455/config.txt)

**Reply by jprestop** [reply]
Thank you @jcorner1.

Surprisingly, I still see the following in your config.log file:
```
/usr/bin/ld: cannot find -lhdf5_hl
/usr/bin/ld: cannot find -lhdf5
/usr/bin/ld: cannot find -lz
collect2: error: ld returned 1 exit status
```
So, let's take a step back.  I see that you have the following variables set in your install_met_env.triton file:
```
export MET_GRIB2CLIB=${EXTERNAL_LIBS}/lib
export MET_GRIB2CINC=${EXTERNAL_LIBS}/include
export GRIB2CLIB_NAME=-lgrib2c
export MET_BUFRLIB=${EXTERNAL_LIBS}/lib
export BUFRLIB_NAME=-lbufr
```
However, looking at your output from running "ls /home/jcorner1/Masters/met/MET-11.0.0/external_libs/lib", I do not see the required files there for those libraries.  You only want to set those variables if you already have those libraries built and do not want to recompile them.  Despite you having some libraries built and others not built, please go ahead and comment out everything below:
```
export EXTERNAL_LIBS=${TEST_BASE}/external_libs
``` 
in your install_met_env.triton and then try recompiling using the script.  It should then build all of the required libraries and hopefully MET too.

And one other correction, which is totally on me (my apologies again).  Please change the following:
```
export MET_PYTHON_LD="-L/anaconda3/lib/python3.9/config-3.9-x86_64-linux-gnu -L/anaconda3/lib  -l python3.9 -lcrypt -lpthread -ldl  -lutil -lm"
```
to the following:
```
export MET_PYTHON_LD="-L/anaconda3/lib/python3.9/config-3.9-x86_64-linux-gnu -L/anaconda3/lib  -lpython3.9 -lcrypt -lpthread -ldl  -lutil -lm"
```
I accidentally put a space between "-l" and "python3.9" and that will not work.

Please let me know how that goes.  Thanks!

**Reply by jcorner1** [reply]
hi @jprestop,

here is the screen output when I ran the new screen:

```
[jcorner1@triton met]$ ./compile_MET_all.sh install_met_env.triton 

TEST_BASE = /home/jcorner1/Masters/met/MET-11.0.0
COMPILER  = gnu
MET_SUBDIR = /home/jcorner1/Masters/met/MET-11.0.0
MET_TARBALL = MET-11.0.0.tar.gz
USE_MODULES = FALSE
LD_LIBRARY_PATH = /home/jcorner1/Masters/met/MET-11.0.0/external_libs/lib:/anaconda3/lib

Compiling libraries into: /home/jcorner1/Masters/met/MET-11.0.0/external_libs

USE_MODULES = FALSE

export  CC=/usr/bin/gcc
export CXX=/usr/bin/g++
export  FC=/usr/bin/gfortran
export F77=/usr/bin/gfortran
export F90=/usr/bin/gfortran

Using python version: 
Python 3.9.12

Compiling GSL_2.1 at Thu Jan 19 21:21:04 UTC 2023
cd /home/jcorner1/Masters/met/MET-11.0.0/external_libs/gsl/gsl-2.1
./configure --prefix=/home/jcorner1/Masters/met/MET-11.0.0/external_libs > configure.log 2>&1
make > make.log 2>&1
make install > make_install.log 2>&1

Compiling BUFRLIB_v11.3.0 at Thu Jan 19 21:26:38 UTC 2023
cd /home/jcorner1/Masters/met/MET-11.0.0/external_libs/bufrlib/BUFRLIB_v11.3.0

Compiling ZLIB at Thu Jan 19 21:27:02 UTC 2023
cd /home/jcorner1/Masters/met/MET-11.0.0/external_libs/zlib/zlib-1.2.11
./configure --prefix=/home/jcorner1/Masters/met/MET-11.0.0/external_libs > configure.log 2>&1
make > make.log 2>&1
make install > make_install.log 2>&1
rm /home/jcorner1/Masters/met/MET-11.0.0/external_libs/lib/zlib.a

Compiling LIBPNG at Thu Jan 19 21:27:13 UTC 2023
cd /home/jcorner1/Masters/met/MET-11.0.0/external_libs/libpng/libpng-1.6.37
./configure --prefix=/home/jcorner1/Masters/met/MET-11.0.0/external_libs LDFLAGS=-L/home/jcorner1/Masters/met/MET-11.0.0/external_libs/lib CPPFLAGS=-I/home/jcorner1/Masters/met/MET-11.0.0/external_libs/include > configure.log 2>&1
make > make.log 2>&1
make install > make_install.log 2>&1

Compiling JASPER at Thu Jan 19 21:27:49 UTC 2023
cd /home/jcorner1/Masters/met/MET-11.0.0/external_libs/jasper/jasper-1.900.1
./configure --prefix=/home/jcorner1/Masters/met/MET-11.0.0/external_libs > configure.log 2>&1
make > make.log 2>&1
make install > make_install.log 2>&1

Compiling G2CLIB at Thu Jan 19 21:28:57 UTC 2023
cd /home/jcorner1/Masters/met/MET-11.0.0/external_libs/g2clib/g2clib-1.6.0
make > make.log 2>&1

Compiling HDF5 at Thu Jan 19 21:29:07 UTC 2023
cd /home/jcorner1/Masters/met/MET-11.0.0/external_libs/hdf5/hdf5-1.12.2
./configure --prefix=/home/jcorner1/Masters/met/MET-11.0.0/external_libs --with-zlib=/home/jcorner1/Masters/met/MET-11.0.0/external_libs/lib CFLAGS=-fPIC CXXFLAGS=-fPIC FFLAGS=-fPIC LDFLAGS=-L/home/jcorner1/Masters/met/MET-11.0.0/external_libs/lib CPPFLAGS=-I/home/jcorner1/Masters/met/MET-11.0.0/external_libs/include > configure.log 2>&1
make install > make_install.log 2>&1

Compiling NetCDF-C at Thu Jan 19 21:48:13 UTC 2023
cd /home/jcorner1/Masters/met/MET-11.0.0/external_libs/netcdf/netcdf-c-4.7.4
./configure --prefix=/home/jcorner1/Masters/met/MET-11.0.0/external_libs CFLAGS=-fPIC CXXFLAGS=-fPIC LDFLAGS=-L/home/jcorner1/Masters/met/MET-11.0.0/external_libs/lib CPPFLAGS=-I/home/jcorner1/Masters/met/MET-11.0.0/external_libs/include > configure.log 2>&1
configure returned with non-zero (1) status
[jcorner1@triton met]$ 
```
I'm not sure what configure, make, or install log file to send as I'm not seeing the same ones (or just forgot the location) files I saw the first time I compiled MET. 

[install_met_env.triton.txt](https://github.com/dtcenter/METplus/files/10461490/install_met_env.triton.txt)

**Reply by jprestop** [reply]
HI @jcorner1.

First, and my apologies for not catching these earlier, I see that you have the following set:
```
export COMPILER=gnu
```

Could you please change that to the following?
```
export COMPILER=gnu_4.8.5
```

I think that BUFRLIB may not have compiled successfully as this [section](https://github.com/dtcenter/MET/blob/681dec568ed40bd64d7df63230b8c41df12bda18/internal/scripts/installation/compile_MET_all.sh#L381) of code needs to know which version of the GNU compiler you're using in order to know which flags to use.

Also, I see that the following is set:
```
export MET_TARBALL=MET-11.0.0.tar.gz
```
However, the tarfile is named v11.0.0.tar.gz when you run the following command:
```
https://github.com/dtcenter/MET/archive/refs/tags/v11.0.0.tar.gz
```
Unless you changed the name of the file after downloading, your config file should contain:
```
export MET_TARBALL=v11.0.0.tar.gz
```

So, please change those (MET_TARBALL, if necessary) and rerun the script.

Next, I see that NetCDF-C had trouble:
```
Compiling NetCDF-C at Thu Jan 19 21:48:13 UTC 2023
cd /home/jcorner1/Masters/met/MET-11.0.0/external_libs/netcdf/netcdf-c-4.7.4
./configure --prefix=/home/jcorner1/Masters/met/MET-11.0.0/external_libs CFLAGS=-fPIC CXXFLAGS=-fPIC LDFLAGS=-L/home/jcorner1/Masters/met/MET-11.0.0/external_libs/lib CPPFLAGS=-I/home/jcorner1/Masters/met/MET-11.0.0/external_libs/include > configure.log 2>&1
configure returned with non-zero (1) status
```

If NetCDF-C still has trouble after the rerun of the script, could you please pass along the NetCDF-C config.log file?  I believe it will be located at:
/home/jcorner1/Masters/met/MET-11.0.0/external_libs/netcdf/netcdf-c-4.7.4

> I'm not sure what configure, make, or install log file to send as I'm not seeing the same ones (or just forgot the location) files I saw the first time I compiled MET.

MET did not attempt to build due to these issues above, which is why you're not seeing what you saw before.   Once we get these library dependency problems resolved, MET will attempt to build. 

Thanks!

**Reply by jcorner1** [reply]
Hi @jprestop,

I made the above changes and still am having the issue with netcdf. I will send the log files here. 

[configure.txt](https://github.com/dtcenter/METplus/files/10482367/configure.txt)
[config.txt](https://github.com/dtcenter/METplus/files/10482368/config.txt)

**Reply by jprestop** [reply]
HI @jcorner1.

Thank you.  I see the following in the config.log file:
```
configure:5774: error: curl required for remote access. Install curl or build with --disable-dap.
```
It looks like the curl library is needed.  The curl library is frequently already installed systems, however, it seems to be missing on yours, or at least, NetCDF can't seem to find it.  You could try running "which curl", to see if the curl executable is on your system.  If a path comes back (e.g. /usr/bin/curl), in which case we could try to find the library file and tell NetCDF where it is.  Or, you could install the curl package (https://curl.se/download.html) and install it in the /home/jcorner1/Masters/met/MET-11.0.0/external_libs.  Please let me know how you'd like to proceed and if you need any assistance.

**Reply by jcorner1** [reply]
hi @jprestop, 

After running 'which curl', it did find a path to curl '/anaconda3/bin/curl' so it seems the best action would be to point netCDF to that file. Unsure of how to do so, so I will need your assistance. Thanks again!

**Reply by jprestop** [reply]
Hi @jcorner1.  Thank you.  Can you please run the following and send the output?
```
ls -l /anaconda3/lib/libcu*
```

Julie

**Reply by jcorner1** [reply]
Hi @jprestop, Here is the output below:

```
lrwxrwxrwx. 1 root root     16 Jul 19  2022 /anaconda3/lib/libcurl.so -> libcurl.so.4.7.0
lrwxrwxrwx. 1 root root     16 Jul 19  2022 /anaconda3/lib/libcurl.so.4 -> libcurl.so.4.7.0
-rwxrwxr-x. 1 root root 720832 Jul 19  2022 /anaconda3/lib/libcurl.so.4.7.0
```
Thanks!

**Reply by jprestop** [reply]
Hi @jcorner1.  Ok, let's try having you add the following line in your install_met_env.triton file.  It shouldn't really matter where you add it.
```
export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/anaconda3/lib
```
Then please rerun the script and let us know how it goes.

**Reply by jcorner1** [reply]
Hi @jprestop!

It looks like something else is also messing with netcdf still. I'll send the output logs for help further diagnosing the issue. Trying to learn to better understand things, I did look at see that gcc was an issue. I remember that being an issue the first time I tried to install MET, however explicitly setting compiler info within the installation file as shown below didn't change anything. 

```
export  CC=/usr/bin/gcc
export CXX=/usr/bin/g++
export  FC=/usr/bin/gfortran
export F77=/usr/bin/gfortran
export F90=/usr/bin/gfortran
```

Thanks for helping to get the issue resolved!

[config.txt](https://github.com/dtcenter/METplus/files/10484368/config.txt)
[configure.txt](https://github.com/dtcenter/METplus/files/10484369/configure.txt)

**Reply by jprestop** [reply]
Hi @jcorner1.

I see the same problem with curl in the config.log file. Let's try to get that fixed.

My apologies - I asked you to set LD_LIBRARY_PATH, but that is used at runtime.    Instead, I should have had you try adding to LDFLAGS and CPPFLAGS.

Please set the following in your install_met_env.triton file:
```
export CURL_LIB_DIR=/anaconda3/lib
export CURL_INC_DIR=/anaconda3/include
```

Then, please change [this line](https://github.com/dtcenter/MET/blob/681dec568ed40bd64d7df63230b8c41df12bda18/internal/scripts/installation/compile_MET_all.sh#L627) of the compile_MET_all.sh script from:
```
./configure --prefix=${LIB_DIR} --with-zlib=${LIB_Z} CFLAGS=-fPIC CXXFLAGS=-fPIC FFLAGS=-fPIC LDFLAGS=-L${LIB_DIR}/lib:${LIB_Z} CPPFLAGS=-I${LIB_DIR}/include > configure.log 2>&1
```
to
```
./configure --prefix=${LIB_DIR} --with-zlib=${LIB_Z} CFLAGS=-fPIC CXXFLAGS=-fPIC FFLAGS=-fPIC LDFLAGS=-L${LIB_DIR}/lib:${LIB_Z}:${CURL_LIB_DIR} CPPFLAGS=-I${LIB_DIR}/include:${CURL_INC_DIR} > configure.log 2>&1
```

Then, please rerun the script and let us know how it goes.

**Reply by jcorner1** [reply]
Hi @jprestop,

Still seems to be running into some issues with curl. I'll send my files along. 

[install_met_env.triton.txt](https://github.com/dtcenter/METplus/files/10486004/install_met_env.triton.txt)
[config.txt](https://github.com/dtcenter/METplus/files/10486005/config.txt)
[configure.txt](https://github.com/dtcenter/METplus/files/10486006/configure.txt)
[compile_MET_all.sh.txt](https://github.com/dtcenter/METplus/files/10486007/compile_MET_all.sh.txt)

**Reply by jprestop** [reply]
HI @jcorner1.

It looks like my link went to the correct line, but my copy and paste was of the wrong line.  The modified line in your compile_MET_all.sh.txt file was under the HDF5 compilation, but I had intended it to be under the NetCDF-C compilation.  I have modified the file and am sending it back to you.

Please try running with this modified script and let us know how it goes.

[compile_MET_all.sh.new.txt](https://github.com/dtcenter/METplus/files/10486056/compile_MET_all.sh.new.txt)

**Reply by jcorner1** [reply]
Hi @jprestop,

First of all, I would like to say thank you so much for working well past any normal time as that is not necessary. Still seems to be some issue with curl however.

[configure.txt](https://github.com/dtcenter/METplus/files/10486195/configure.txt)
[config.txt](https://github.com/dtcenter/METplus/files/10486196/config.txt)

**Reply by jprestop** [reply]
Hi @jcorner1.  Thank you!

Let's have you try compiling curl with the following commands:
```
cd /home/jcorner1/Masters/met/MET-11.0.0/external_libs
mkdir curl
cd curl
wget https://curl.se/download/curl-7.87.0.tar.gz
tar -zxf curl-7.87.0.tar.gz
cd curl-7.87.0
./configure prefix=/home/jcorner1/Masters/met/MET-11.0.0/external_libs --without-ssl
make install > & make_install.log
```

Then, please remove the CURL_LIB_DIR and CURL_INC_DIR lines from your install_met_env.triton file and use the regular (pre-modifications) compile_MET_all.sh script to compile, which I am reattaching below.  Please let us know how that goes. Just to let you know, I have a training session this morning/early afternoon and may not be able to respond until later.   Hopefully, this will work well!
[compile_MET_all.txt](https://github.com/dtcenter/METplus/files/10491594/compile_MET_all.txt)

**Reply by jcorner1** [reply]
Hi @jprestop

That looks to have done it. MET looks to be successfully installed now. Thanks for all the help, I owe you and other members of DTC many thanks!!!

**Reply by jprestop** [reply]
Hi @jcorner1.  That's great news!  Thank you for letting us know.  It's our pleasure to help.  We want the process of getting up and running to be as quick as possible.
